gtfs/__main__.py

```diff
@@ -1,25 +1,61 @@
-import argparse
-import getpass
+import typer
+from typing_extensions import Annotated
+from chalky import chain
 import logging
 import sys
-
-from prettytable import PrettyTable
-
-from gtfs.utils.FeedSource import FeedSource
-import feed_sources
-# import all the available feed sources
-# pylint: disable=I0011,wildcard-import
-from feed_sources import *
+from prettytable.colortable import ColorTable, Themes
+from halo import Halo
+from .feed_source import FeedSource
+from .feed_sources import __all__ as feed_sources
 
 logging.basicConfig()
 LOG = logging.getLogger()
 LOG.setLevel(logging.INFO)
+app = typer.Typer()
 
-def fetch_all(sources=None):
-    """Fetch from all FeedSources in the feed_sources directory.
-
+def check_predicate(value: str):
+    if value not in ['intersects', 'contains']:
+        raise typer.BadParameter('predicate must be one of intersects or contains')
+    return value
+            
+@app.command()
+def list_feeds(bbox: Annotated[str, typer.Argument(help="pass value as a string separated by commas like this: xmin,ymin,xmax,ymax ")],
+    predicate: Annotated[str, typer.Option("--predicate", "-p", help="intersects or contains", case_sensitive=True, callback=check_predicate)] = 'intersects'):
+    """Filter feeds spatially based on bounding box."""
+    if ',' not in bbox or len(bbox.split(',')) != 4:
+        LOG.critical(chain.bright_red | "Please pass bbox as a string separated by commas like this: xmin,ymin,xmax,ymax")
+        sys.exit(1)
+    for coord in bbox.split(','):
+        try:
+            float(coord)
+        except ValueError:
+            LOG.critical(chain.bright_red | "Please pass only numbers as bbox values!")
+            sys.exit(1)
+    spinner = Halo(text='Filtering...', text_color="cyan", spinner='dots')
+    spinner.start()
+    xmin, ymin, xmax, ymax = [float(coord) for coord in bbox.split(',')]
+    ptable = ColorTable(['Feed Source', 'Transit URL', 'Bounding Box'], theme=Themes.OCEAN)
+    for src in feed_sources:
+        inst = src()
+        feed_bbox = inst.bbox
+        feed_xmin, feed_ymin, feed_xmax, feed_ymax = feed_bbox
+        if predicate == 'contains':
+            if (feed_xmin >= xmin and feed_xmin < xmax) and (feed_ymin >= ymin and feed_ymin < ymax) and (feed_xmax <= xmax and feed_xmax > xmin) and (feed_ymax <= ymax and feed_ymax > ymin):
+                row=[src, chain.bright_yellow | inst.url , feed_bbox ]
+                ptable.add_row(row)
+        else:
+            # intersects
+            if (feed_xmin >= xmin and feed_xmin <= xmax) or (feed_ymin >= ymin and feed_ymin <= ymax) or (feed_xmax <= xmax and feed_xmax >= xmin) or (feed_ymax <= ymax and feed_ymax >= ymin) or (xmin >= feed_xmin and xmin <= feed_xmax) or (ymin >= feed_ymin and ymin <= feed_ymax) or (xmax <= feed_xmax and xmax >= feed_xmin) or (ymax <= feed_ymax and ymax >= feed_ymin):
+                row=[src, chain.bright_yellow | inst.url , feed_bbox ]
+                ptable.add_row(row)
```

Comment:
this is unreadable haha

I'd do smth like `utils.geom` and a `bbox_contains_bbox` function and then a `bbox_intersects_bbox` function. also I'm quite sure the intersects function isn't quite right, but we'll see when testing.

Reply:
I too was unsure about the intersects function....I will try to check it again and I will also create a separate util for both..

Reply:
I was shocked when u didn't say anything about it yesterday haha